src/app/reporting/Engine.cpp

```diff
@@ -139,14 +150,27 @@ CHIP_ERROR Engine::BuildSingleReportDataAttributeDataList(ReportData::Builder &
             }
         }
     }
-    attributeDataList.EndOfAttributeDataList();
-    err = attributeDataList.GetError();
-
 exit:
+    if ((err == CHIP_ERROR_BUFFER_TOO_SMALL) || (err == CHIP_ERROR_NO_MEMORY))
+    {
+        ChipLogDetail(DataManagement, "<RE:Run> We cannot put more chunks into this report. Enable chunking.");
+        err = CHIP_NO_ERROR;
```

Comment:
This would mean that the caller of this function will continue trying to pack in more attributes/events, when that's not what you want.

More importantly, you actually need to keep track of which `ReadHandler` and path you last left off at so that you can resume at the **same** read handler AND path. We do not want to service another read handler till we've finished the last one that couldn't get completed.

Reply:
The iteration state is recorded in the path iterator in the ReadHandler, so it should be fine.
If our peer is pretty slow, of course we can serve another peer.

Agree on we have lots of things to do on scheduling the generating of reports.